ipn/ipnlocal/drive.go

```diff
@@ -343,15 +335,29 @@ func (b *LocalBackend) driveRemotesFromPeers(nm *netmap.NetworkMap) []*drive.Rem
 
 				for _, candidate := range latestNetMap.Peers {
 					if candidate.ID() == peerID {
-						online := candidate.Online()
+						// Exclude offline peers.
 						// TODO(oxtoacart): for some reason, this correctly
 						// catches when a node goes from offline to online,
 						// but not the other way around...
-						return online != nil && *online
+						online := candidate.Online()
+						if online == nil || !*online {
+							return false
+						}
+
+						// Check that the peer is allowed to share with us.
+						for _, addr := range candidate.Addresses().AsSlice() {
+							capsMap := b.PeerCaps(addr.Addr())
+							_, ok := capsMap[tailcfg.PeerCapabilityTaildriveSharer]
+							if ok {
+								return true
+							}
+						}
```

Comment:
@bradfitz didn't we add a cheaper version of this?

Reply:
Oh, I meant to take out the AsSlice(). One moment ...

Reply:
Fixed.